src/components/tl/mlx5/mcast/tl_mlx5_mcast.h

```diff
@@ -40,6 +41,24 @@
 #define ONE_SIDED_ASYNCHRONOUS_PROTO        2
 #define ONE_SIDED_SLOTS_COUNT               2                /* number of memory slots during async design */
 #define ONE_SIDED_SLOTS_INFO_SIZE           sizeof(uint32_t) /* size of metadata prepended to each slots in bytes */
+#define ONE_SIDED_INVALID                  -1
+#define ONE_SIDED_VALID                    -2
+#define ONE_SIDED_PENDING_INFO             -3
+#define ONE_SIDED_PENDING_DATA             -4
+#define ONE_SIDED_MAX_ALLGATHER_COUNTER     32
+#define ONE_SIDED_MAX_CONCURRENT_LEVEL      64
+
+/* 32 here is the bit count of ib send immediate */
+#define ONE_SIDED_MAX_PACKET_COUNT(_max_count)                              \
```

Comment:
Since this macro is used only in a single file, I will move the macro def to the same file. However, I rather keep the def separate. This is because in the future we most probably will call this macro in other places too.

Reply:
> This is because in the future we most probably will call this macro in other places too.

This macro is adding unnecessary complexity to the code. Complexity should be added once it is used. I leave it up to you but still consider this is a relevant comment.

Besides, why not making this macro a function?

Reply:
Sure, thanks for the comment. Let me change it so that we use a function instead of a macro.